pg_shard.c

```diff
@@ -226,8 +226,18 @@ PgShardPlanner(Query *query, int cursorOptions, ParamListInfo boundParams)
 
 		ErrorIfQueryNotSupported(distributedQuery);
 
-		/* compute the list of shards this query needs to access */
+		/*
+		 * Compute the list of shards this query needs to access.
+		 * Error out if there are no existing shards for the table.
+		 */
 		queryShardList = DistributedQueryShardList(distributedQuery);
+		if (queryShardList == NIL)
+		{
+			ereport(ERROR, (errmsg("cannot plan SELECT query"),
```

Comment:
I think we can use `ERRCODE_OBJECT_NOT_IN_PREREQUISITE_STATE` here: we have an object (the distributed table) which hasn't been initialized yet (by creating shards). This code is used in PostgreSQL for other analogous situations, such as not calling `nextval` for a sequence before asking it what its current value is (`grep` for it).